geoportailv3/static/js/layerpermalinkservice.js

```diff
@@ -0,0 +1,254 @@
+/**
+ * @fileoverview This files defines an Angular Service for managing the
+ * selected layers permalinks
+ */
+
+goog.provide('app.LayerPermalinkManager');
+
+goog.require('app');
+goog.require('app.GetLayerForCatalogNode');
+goog.require('app.StateManager');
+goog.require('app.Themes');
+goog.require('ngeo.BackgroundLayerMgr');
+
+
+
+/**
+ * @constructor
+ * @param {app.StateManager} appStateManager
+ * @param {ngeo.BackgroundLayerMgr} ngeoBackgroundLayerMgr
+ * @param {app.GetLayerForCatalogNode} appGetLayerForCatalogNode
+ * @param {app.Themes} appThemes
+ * @ngInject
+ */
+app.LayerPermalinkManager =
+    function(appStateManager, ngeoBackgroundLayerMgr,
+        appGetLayerForCatalogNode, appThemes) {
+  /**
+   * @type {app.Themes}
+   * @private
+   */
+  this.appThemes_ = appThemes;
+
+  /**
+   * @type {app.StateManager}
+   * @private
+   */
+  this.stateManager_ = appStateManager;
+
+  /**
+   * @type {ngeo.BackgroundLayerMgr}
+   * @private
+   */
+  this.backgroundLayerMgr_ = ngeoBackgroundLayerMgr;
+
+  /**
+   * @type {app.GetLayerForCatalogNode}
+   * @private
+   */
+  this.getLayerFunc_ = appGetLayerForCatalogNode;
+};
+
+
+/**
+ * @param {Array.<ol.layer.Layer>} layers
+ * @private
+ */
+app.LayerPermalinkManager.prototype.setLayerState_ = function(layers) {
+  var layerIds = goog.array.map(layers, function(layer) {
+    return layer.get('queryable_id');
```

Comment:
We still don't have ids for layers? The only ids we have are the `queryable_id`s?